### DISCUSSION TITLE
Licensing

### DISCUSSION THREAD

**Discussion by goaaats** [goatcorp/Dalamud#534] [Decisions]
Dalamud is currently licensed under AGPLv1.

It would be nice if we could update this to AGPLv3 or later.

To consider:

What needs to be done? How can we reach out to contributors?
Should we consider a clause to prevent commercial use of any kind? Does A/GPL allow for this?
Comments would be appreciated.

**Comment by daemitus** [comment]
How does this license and/or our current license affect closed source plugins?

**Reply by goaaats** [reply]
This technically is already prohibited by A/GPL.

We would have to switch to LGPL which allows for dynamic linking from closed source software, which is fine by me.

**Reply by daemitus** [reply]
Given we already have one "authorized" closed source plugin, we should either switch, or maybe carve out an exception of some kind.

**Reply by VariableVixen** [reply]
Alternatively - and there may well be reasons you don't want to do it, I dunno - you could also forbid closed-source plugins. I'm not trying to favour one or the other, just saying that it's another option.

**Reply by daemitus** [reply]
I think we need to  document what our goals are with this, and fit a license to that.

1. Don't distribute Dalamud outside of testing features or providing to a pre-approved audience. (the bluefissure exception)
2. You may link to Dalamud to build plugins
3. You may distribute plugins however you want.
4. We really don't care what you do with said plugins.

**Reply by goaaats** [reply]
I care about:

1. Plugins being able to link to Dalamud(whether it be only open source, or also closed source).
2. If feasible, devs not being able to sell plugins commercially(donations don't count).

This means that we can probably choose between GPLv3/LGPLv3(or later) with an exception re: point 2.

If we decide to go with GPL, we need to take care of the one closed source plugin I can think of.

**Reply by daemitus** [reply]
Ok. so after reading around:
LGPLv3 looks like it will work. [this](https://www.gnu.org/licenses/gpl-faq.en.html#LGPLStaticVsDynamic) explains how dynamic linking applies, and in our case would allow for closed source plugins. Otherwise, we can include the [classpath exception](https://en.wikipedia.org/wiki/GPL_linking_exception#The_Classpath_exception) which will definitely cover it.

Also: when are programs and plugins considered a single source: https://www.gnu.org/licenses/gpl-faq.html#GPLPlugins . I would say we fall under para 2, that the plugin shares complex data structures (sometimes), and are as such a single program. So the LGPL non-commercial clause would cover plugins. We would then just need to find a non-commercial but donation allowed clause for plugins.

**Reply by daemitus** [reply]
correction: there really isnt a non-commercial clause, its just that you have to OSS if you modify or statically link, so there isnt much purpose in commercial usage. so we'd need a non-commercial clause of our own regardless.

**Comment by goaaats** [comment]
I've prepared a draft of a new license for Dalamud.

* It is a verbatim copy of the GNU Lesser General Public License v3.0 which allows for both open and closed source plugins with no restrictions.
* `Addendum A` has been added that excludes commercial use of Dalamud and derivatives/dependants, excluding donations that don't represent an exchange of licensed material.

Please leave a comment here if you have any concerns or questions.

https://gist.github.com/goaaats/8ddb518b9cf0b054c34e5e132fc2d2be
